packages/browser/src/provider.ts

```diff
@@ -0,0 +1,268 @@
+import type { BitskiEngine } from 'bitski-provider';
+import type { BitskiSDK, ProviderOptions } from './-private/sdk';
+import { ProviderError } from 'bitski-provider/lib/errors/provider-error';
+
+interface RequestArguments {
+  method: string;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  params?: unknown[] | object;
+}
+
+interface Web3Provider {
+  currentBlock?: BufferBlock;
+  supportsSubscriptions(): boolean;
+  subscribe(
+    subscribeMethod: string | undefined,
+    subscriptionMethod: string,
+    parameters: any[],
+  ): Promise<string>;
+  unsubscribe(subscriptionId: string, unsubscribeMethod?: string): Promise<boolean>;
+  isRunning(): boolean;
+  start(): void;
+  stop(): void;
+  addProvider(source: Subprovider, index?: number): void;
+  removeProvider(source: Subprovider): void;
+  send(method: string, params: any[]): Promise<any>;
+  sendAsync(payload: JSONRPCRequest | JSONRPCRequest[], cb: JSONRPCResponseHandler): void;
+
+  // Event Emitter API
+  addListener(event: string | symbol, listener: (...args: any[]) => void): this;
+  on(event: string | symbol, listener: (...args: any[]) => void): this;
+  once(event: string | symbol, listener: (...args: any[]) => void): this;
+  removeListener(event: string | symbol, listener: (...args: any[]) => void): this;
+  off(event: string | symbol, listener: (...args: any[]) => void): this;
+  removeAllListeners(event?: string | symbol): this;
+  setMaxListeners(n: number): this;
+  getMaxListeners(): number;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  listeners(event: string | symbol): Function[];
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  rawListeners(event: string | symbol): Function[];
+  emit(event: string | symbol, ...args: any[]): boolean;
+  listenerCount(type: string | symbol): number;
+  // Added in Node 6...
+  prependListener(event: string | symbol, listener: (...args: any[]) => void): this;
+  prependOnceListener(event: string | symbol, listener: (...args: any[]) => void): this;
+  eventNames(): Array<string | symbol>;
+}
+
+interface AddEthereumChainParameter {
+  chainId: string;
+  blockExplorerUrls?: string[];
+  chainName?: string;
+  iconUrls?: string[];
+  nativeCurrency?: {
+    name: string;
+    symbol: string;
+    decimals: number;
+  };
+  rpcUrls?: string[];
+}
+
+interface SwitchEthereumChainParameter {
+  chainId: string; // A 0x-prefixed hexadecimal string
+}
+
+import {
+  type Network,
+  BinanceSmartChain,
+  BinanceSmartChainTestnet,
+  Mainnet,
+  Goerli,
+  Polygon,
+  Mumbai,
+} from 'bitski-provider';
+
+const NETWORK_MAP: Record<string, Network> = {
+  '0x1': Mainnet,
+  '0x5': Goerli,
+  '0x89': Polygon,
+  '0x13881': Mumbai,
+  '0x38': BinanceSmartChain,
+  '0x61': BinanceSmartChainTestnet,
+};
+
+export class BitskiProvider extends EventEmitter implements Web3Provider {
+  private sdkPromise: Promise<Pick<BitskiSDK, 'createProvider'> | null>;
```

Comment:
Does this make more sense as `providerPromise` instead, given it's a subset of the SDK?

Reply:
It's a subset, but it's not actually a provider, it's more like `providerFactoryPromise` but I don't wanna get all Java-y in here 😛